### ISSUE TITLE
Compiling Detours on MinGW

### ISSUE THREAD

**Issue by sgarfinkel** [microsoft/Detours#68]
I'm pretty close to getting this to work. I had to remove some MSVC-specific cruft from the header file to work around some duplicated typedefs when _MSVC_VER is undefined.

The only remaining sticking point is dealing with SEH calls. MinGW does not port these, and I'm unsure what the workaround might be.

For reference, these are the errors:
```
detours.cpp: In function 'bool detour_is_imported(PBYTE, PBYTE)':
detours.cpp:44:5: error: '__try' was not declared in this scope
     __try {
     ^~~~~
detours.cpp:68:5: error: '__except' was not declared in this scope
     __except(GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ?
     ^~~~~~~~
detours.cpp:68:5: note: suggested alternative: 'accept'
     __except(GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ?
     ^~~~~~~~
     accept
```

I'm trying to compile under MinGW to support interoperability with Kotlin, which can only compile under MinGW. Although MinGW has some support for interoperability with binaries compiled under MSVC, there are still issues with undefined symbols if you go that route.

Has anyone successfully compiled Detours with MinGW? If so what was your workaround for this?

**Comment by alf-p-steinbach** [comment]
I'm at the point of trying to use Detours with MinGW g++ and my immediate hunch is that it will work to use Visual C++ to create a DLL that will be usable from g++.

**Comment by bgianfo** [comment]
Compilation issues were fixed in #108, so it should compile cleanly for you now. 
We can't claim to support this configuration though.
